internal/uvm/share.go

```diff
@@ -0,0 +1,75 @@
+package uvm
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// Share shares in file(s) from `reqHostPath` on the host machine to `reqUVMPath` inside the UVM.
+// This function handles both LCOW and WCOW scenarios.
+func (uvm *UtilityVM) Share(ctx context.Context, reqHostPath, reqUVMPath string, readOnly bool) (retErr error) {
+	if uvm.OS() == "windows" {
+		if _, err := uvm.GetVSMBUvmPath(ctx, reqHostPath, readOnly); err == ErrNotAttached {
+			// share file has not been added yet, add it now
+			options := uvm.DefaultVSMBOptions(readOnly)
+			vsmbShare, err := uvm.AddVSMB(ctx, reqHostPath, options)
```

Comment:
We should probably always call `AddVSMB`, even if the share is already added. If we want to support removing a share in the future, we will want actual calls to `AddVSMB` each time, so the ref-counting works out.

Reply:
Good point. Wrote this a while ago so I'm not sure why I added that extra check, but removed.